### DISCUSSION TITLE
Install ok but doesnt work

### DISCUSSION THREAD

**Discussion by 42corsair** [sebr/bhyve-home-assistant#110] [General]
Getting the following error when I check the configuration or try to restart:

```
Platform error binary_sensor.bhyve - cannot import name 'Client' from 'custom_components.bhyve.pybhyve' (unknown location)
Component error: bhyve - cannot import name 'Client' from 'custom_components.bhyve.pybhyve' (unknown location)
Platform error sensor.bhyve - cannot import name 'Client' from 'custom_components.bhyve.pybhyve' (unknown location)
Platform error switch.bhyve - cannot import name 'Client' from 'custom_components.bhyve.pybhyve' (unknown location)
```

Version
core-2021.11.5
Version
supervisor-2022.01.1
Operating System
Home Assistant OS 7.4

configuration.yaml (added to end)

```yaml
bhyve:
  username: !secret bhyve_username
  password: !secret bhyve_password

sensor:
  - platform: bhyve

switch:
  - platform: bhyve

binary_sensor:
  - platform: bhyve
```

Have tried upgrading to latest versions of everything, then downgraded back down to core-2021.11.5 to try and same issue...

Any ideas?

**Comment by sebr** [comment]
Which version of the b-hyve integration are you using?

**Comment by 42corsair** [comment]
2.2.1 downloaded with HACS

**Comment by 42corsair** [comment]
It never shows in the HACS area and never becomes available as an Integration elsewhere, but all the files seem like they are there in the directory - if that helps at all

**Comment by sebr** [comment]
I think there might have been some corruption in the install, as I can't see any reason for these errors. I would suggest that you delete everything in `custom_components/bhyve`, restart HA, then reinstall with HACS

**Reply by 42corsair** [reply]
Ok, I will give it a try - thanks!  Does HACS do anything other than unzip everything from the archive into the folder?  Maybe I can just do it manually and it would work the same?

**Reply by sebr** [reply]
Yes - it also registers the installation to provide extra functionality: checking for updates, uninstall/reinstall etc.

Otherwise, not really :)

**Reply by 42corsair** [reply]
Do you have manual install instructions?  HACS does not seem to work and now I am being rate limited by Github so it's disabled according to the message... :(
I'm going to try just unziping the archive into the folder for now...  thanks for you assistance!

**Comment by 42corsair** [comment]
I deleted everything in the bhyve folder, unziped the bhyve-home-assistant-2.2.1.zip, copied everything into the bhyve folder and still getting the same message...  it's gotta be something simple I am missing here?

**Reply by sebr** [reply]
Can you turn on bhyve debug logs by adding

```yaml
logger:
  logs:
    custom_components.bhyve: debug
    pybhyve: debug
```

and then provide more complete log from `home-assistant.log`?

thx

**Comment by 42corsair** [comment]
I got it working - last thing I tried clicking ignore on a bunch on things HA had detected that I wasnt using - then the configuration error went away - and all the sensors began to show up!

**Comment by 42corsair** [comment]
Thank you for the help and quick response!
